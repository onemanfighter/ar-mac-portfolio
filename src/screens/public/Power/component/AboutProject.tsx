import { Box, Button, Heading, Text } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import packageJson from '../../../../../package.json';

const AboutProject = () => {
  const { t } = useTranslation();
  return (
    <Box>
      <Heading as="h2" size="lg" color="white" textAlign="start" m={5}>
        {t('PowerScreen.sidebar.header')}
        <Text fontSize="sm">
          {t('PowerScreen.sidebar.version', {
            version: packageJson.version,
          })}
        </Text>
      </Heading>
      <Box
        display={'flex'}
        flexDirection={'row'}
        justifyContent={'center'}
        columnGap={'5'}
      >
        <Button
          as={Link}
          to={'https://github.com/onemanfighter/ar-mac-portfolio'}
          target="_blank"
          variant={'outline'}
          size={'sm'}
          colorScheme="yellow"
        >
          {t('PowerScreen.sidebar.button.starOnGithub')}
        </Button>
        <iframe
          src="https://github.com/sponsors/onemanfighter/button"
          title="Sponsor onemanfighter"
          height="32"
          width="114"
        ></iframe>
      </Box>
      <Text
        color="white"
        textAlign="start"
        fontSize="md"
        m={5}
        align={'justify'}
      >
        {t('PowerScreen.sidebar.aboutProject')}
      </Text>
    </Box>
  );
};

export default AboutProject;
