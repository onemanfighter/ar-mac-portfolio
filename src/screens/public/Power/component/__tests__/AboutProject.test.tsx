import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import AboutProject from '../AboutProject';

describe('About project', () => {
  it('should render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <AboutProject />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
