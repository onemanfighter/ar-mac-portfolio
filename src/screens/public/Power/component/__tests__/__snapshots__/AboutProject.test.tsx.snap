// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About project should render correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <h2
      class="chakra-heading css-ggj6r2"
    >
      Apple mac OS simulator
      <p
        class="chakra-text css-g5hezf"
      >
        v0.4.0
      </p>
    </h2>
    <div
      class="css-1q9a503"
    >
      <a
        class="chakra-button css-4xx2wk"
        href="https://github.com/onemanfighter/ar-mac-portfolio"
        target="_blank"
      >
        ⭐️ Star on GitHub
      </a>
      <iframe
        height="32"
        src="https://github.com/sponsors/onemanfighter/button"
        title="Sponsor onemanfighter"
        width="114"
      />
    </div>
    <p
      class="chakra-text css-e72f3b"
    >
      This is an open-source project to simulate the Apple mac OS interface. This project is built using ReactJS. You can contribute to this project by giving a star on GitHub.
    </p>
  </div>
</div>
`;
